pkg/controller/backupinfrastructure/backup_infrastructure_control.go

```diff
@@ -0,0 +1,372 @@
+			if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateSucceeded, gardenv1beta1.ShootLastOperationTypeDelete, "Backup Infrastructure has been successfully deleted.", 100, nil); updateErr != nil {
+				backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after deletion successful: %+v", updateErr)
+				return true, updateErr
+			}
+			if err := c.removeFinalizer(op); err != nil {
+				return true, err
+			}
+			return false, nil
+		}
+	}
+
+	// When a BackupInfrastructure deletion timestamp is not set we need to create/reconcile the backup infrastructure.
+	if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateProcessing, operationType, "Reconciliation of Backup Infrastructure state in progress.", 1, nil); updateErr != nil {
+		backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after reconciliation start: %+v", updateErr)
+		return true, updateErr
+	}
+	if reconcileErr := c.reconcileBackupInfrastructure(op); reconcileErr != nil {
+		if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateError, operationType, reconcileErr.Description+" Operation will be retried.", 1, reconcileErr); updateErr != nil {
+			backupInfrastructureLogger.Errorf("Could not update the BackupInfrastructure status after reconciliation error: %+v", updateErr)
+			return true, updateErr
+		}
+		return true, errors.New(reconcileErr.Description)
+	}
+	if operationType == gardenv1beta1.ShootLastOperationTypeCreate {
+		c.recorder.Eventf(backupInfrastructure, corev1.EventTypeNormal, gardenv1beta1.EventReconciled, "[%s] Reconciled Backup Infrastructure state", operationID)
+	}
+	if updateErr := c.updateBackupInfrastructureStatus(op, gardenv1beta1.ShootLastOperationStateSucceeded, operationType, "Backup Infrastructure has been successfully reconciled.", 100, nil); updateErr != nil {
+		backupInfrastructureLogger.Errorf("Could not update the Shoot status after reconciliation success: %+v", updateErr)
+		return true, updateErr
+	}
+
+	return true, nil
+}
+
+// reconcileBackupInfrastructure reconciles a BackupInfrastructure state.
+func (c *defaultControl) reconcileBackupInfrastructure(o *operation.Operation) *gardenv1beta1.LastError {
+
+	// We create botanists (which will do the actual work).
+	botanist, err := botanistpkg.New(o)
+	if err != nil {
+		return formatError("Failed to create a Botanist", err)
+	}
+	seedCloudBotanist, err := cloudbotanistpkg.New(o, common.CloudPurposeSeed)
+	if err != nil {
+		return formatError("Failed to create a Seed CloudBotanist", err)
+	}
+
+	var (
+		defaultRetry = 30 * time.Second
+
+		f                     = flow.New("Backup Infrastructure creation").SetProgressReporter(o.ReportBackupInfrastructureProgress).SetLogger(o.Logger)
+		deployBackupNamespace = f.AddTask(botanist.DeployBackupNamespaceFromBackupInfrastructure, defaultRetry)
```

Comment:
Change `DeployBackupNamespaceFromBackupInfrastructure` to `DeployBackupNamespace`?

Reply:
Actually there 2 places where we deploy backup namespace, one is in shoot reconcile and other is in backup_infrastructure reconcile. In backupInfrastructure reconcile since operation object is populated with backupInfrastructure I have to get backup namespace name from there.
In case of shoot reconcile (for backward compatibility ) the backup namespace is computed from shoot object populated in operation struct. So, two different implementation of same end result required and hence two different function. On thing i can do is rename it as you suggested which will create namespace from the info from backupInfrastrucutre and keep other function `DeployBackupNamesapceFromShoot` as it is to be used in shoot reconcile. Or even i can move that logic in `MoveTerraformResources` function.